relays/ethereum/src/substrate_client.rs

```diff
@@ -83,12 +83,20 @@ impl Default for SubstrateSigningParams {
 	}
 }
 
+// TODO: Remove
+enum BridgeInstance {
```

Comment:
I think long-term I'd prefer if this was a generic parameter instead of an enum, i.e.:
```rust
trait BridgeInstance {
 fn whatever_parameters_you_need(&self);
}
struct Kovan;
impl BridgeInstance for Kovan {
 fn ...
}
struct Rialto;
impl BridgeInstance of Rialto {
  fn..
}
```
The reason for this is that it will allow using `relay` code as a library and easily implement support for custom chains. I.e. IMHO what we should target with the relay is an architecture with crates like that:
```
- relay-cli // a BINARY - CLI runner for the relay, might list "known" networks
- relay // a LIBRARY - that allows building your own relay, should most likely be generic over some parameters
```
Related: #162 #75 #61 #60

Reply:
Just sharing my thoughts here. I was thinking that we may actually have several relay binaries - one for every pair of chains. We may extract almost everything from relay to shared crate. What should stay in chain-specific binaries are files like: `cli.yml`, `ethereum_client.rs`, `ethreum_deploy_contract.rs`, `substrate_types.rs`, ... So we'll have separate binary for Rialto <-> Kovan, Rialto <-> TestPoA, ... We may achieve same functionality with conditional compilation, though, but it'll be too much for single binary crate imo.

Reply:
I'm not a fan of feature gating either. I like the multi-binary approach as well, pretty much it's another take on the `CLI + Library` split, where we basically have multiple CLI (binaries) ourselves. Probably an advantage of that approach would be that the CLI creation process would be generic/abstract as well, since we would like to keep the code DRY.

Reply:
I didn't really consider the multi-binary approach. The vision I had for this was one where you would configure the relay through the CLI:

```bash
$ relay-cli relay --from ETH --to SUB --instance Kovan
$ relay-cli relay --from KSM --to DOT
$ relay-cli exchange --from KSM --to DOT
```

Although a multi-binary approach might be necessary to support different runtimes. From a user point of view that seems annoying though. 

I like the library + binary split, and maybe that'll ease of the UX around having different binaries for pairs of chains. I'll think about it some more.